doc/build-unix.md

```diff
@@ -163,32 +186,67 @@ mkdir -p $BDB_PREFIX
 # Fetch the source and verify that it is not tampered with
 wget 'https://download.oracle.com/berkeley-db/db-4.8.30.NC.tar.gz'
 echo '12edc0df75bf9abd7f82f821795bcee50f42cb2e5f76a6a281b85732798364ef  db-4.8.30.NC.tar.gz' | sha256sum -c
-# -> db-4.8.30.NC.tar.gz: OK
+# MUST output: db-4.8.30.NC.tar.gz: OK
 tar -xzvf db-4.8.30.NC.tar.gz
 
+# Fetch, verify that it is not tampered with and apply clang related patch
+cd db-4.8.30.NC
+wget 'https://gist.githubusercontent.com/LnL7/5153b251fd525fe15de69b67e63a6075/raw/7778e9364679093a32dec2908656738e16b6bdcb/clang.patch'
+echo '7a9a47b03fd5fb93a16ef42235fa9512db9b0829cfc3bdf90edd3ec1f44d637c clang.patch' | sha256sum -c
```

Comment:
is this a new requirement to build dbd even for gcc?

Reply:
Trying to follow the guide as is here: https://github.com/BitcoinUnlimited/BitcoinUnlimited/blob/release/doc/build-unix.md#berkeley-db

I get this error when running
`test@test-VirtualBox:~/testDB/db-4.8.30.NC/build_unix$ make`

```
In file included from ../dist/../dbinc/mutex_int.h:12,
                 from ../dist/../dbinc/mutex.h:15,
                 from ./db_int.h:884,
                 from ../dist/../cxx/cxx_db.cpp:11:
../dist/../dbinc/atomic.h:179:19: error: definition of ‘int __atomic_compare_exchange(db_atomic_t*, atomic_value_t, atomic_value_t)’ ambiguates built-in declaration ‘bool __atomic_compare_exchange(long unsigned int, volatile void*, void*, void*, int, int)’
  179 | static inline int __atomic_compare_exchange(
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~~
make: *** [Makefile:2018: cxx_db.o] Error 1
```

This is on a clean Ubuntu 19.10, with
```
test@test-VirtualBox:~/testDB/db-4.8.30.NC/build_unix$ gcc --version
gcc (Ubuntu 9.2.1-9ubuntu2) 9.2.1 20191008
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Applying the patch makes it build.

Reply:
I see makes sense, thanks for the explanation. Will do a proper review now.